src/VsExtension/VsEvents/OnBuildPackageRestorer.cs

```diff
@@ -715,8 +715,9 @@ private void LogToVS(VerbosityLevel verbosityLevel, string message)
                 // Switch to main thread to update the progress dialog, output window or error list window
                 await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                // Only show messages with VerbosityLevel.Normal, VerbosityLevel.Minimal or VerbosityLevel.Quiet
-                if (verbosityLevel <= VerbosityLevel.Normal)
+                // Only show messages with VerbosityLevel.Normal. That is, info messages only.
+                // Do not show errors, warnings, verbose or debug messages on the progress dialog
+                if (verbosityLevel == VerbosityLevel.Normal)
```

Comment:
1) The check changed here is for the progress dialog. This does not use the msbuild verbosity at all
2) The statement WriteLine just below is for the output window. This method uses the msbuild verbosity and only writes to the output window if the verbosity of the message being written is equal to or better than msbuild verbosity
3) The last check is for the error list window, which only prints errors and warnings. This does not use the msbuild verbosity either

All the checks above are performed after switching to the main thread.

Even before switching to the main thread, we simply return if the verbosity level is worse than Info  (VerbosityLevel.Normal) AND verbosity of incoming message is worse than msbuild verbosity. Otherwise, we go ahead and switch to the main thread, since, at least one of the progress dialog, or the output window or the error list window will be written to